client/my-sites/media-library/upload-url.jsx

```diff
@@ -13,71 +14,70 @@ import analytics from 'lib/analytics';
 import FormTextInput from 'components/forms/form-text-input';
 import MediaActions from 'lib/media/actions';
 
-export default React.createClass( {
-	displayName: 'MediaLibraryUploadUrl',
-
-	propTypes: {
-		site: React.PropTypes.object,
-		onAddMedia: React.PropTypes.func,
-		onClose: React.PropTypes.func,
-		className: React.PropTypes.string
-	},
-
-	getInitialState() {
-		return {
-			value: '',
-			isError: false
-		};
-	},
-
-	getDefaultProps() {
-		return {
-			onAddMedia: noop,
-			onClose: noop
-		};
-	},
-
-	upload( event ) {
-		var isError = ! event.target.checkValidity();
+class MediaLibraryUploadUrl extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		site: PropTypes.object,
+		onAddMedia: PropTypes.func,
+		onClose: PropTypes.func,
+	};
 
-		this.setState( {
-			isError: isError
-		} );
+	static defaultProps = {
+		onAddMedia: noop,
+		onClose: noop,
+	};
+
+	state = {
+		value: '',
+		isError: false,
+	};
+
+	upload = event => {
+		const isError = ! event.target.checkValidity();
+
+		this.setState( { isError } );
 
 		if ( isError || ! this.props.site ) {
+			event.preventDefault();
```

Comment:
Ah, sorry. So, no matter which flow we take in this function we end up calling `event.preventDefault`. Either here: https://github.com/Automattic/wp-calypso/pull/11822/files#diff-246b58e6f07d7dce525ca439d82eccf9R41 or here: https://github.com/Automattic/wp-calypso/pull/11822/files#diff-246b58e6f07d7dce525ca439d82eccf9R45.
So I thought that this call should be "deduplicated" and moved to a single call, somewhere above https://github.com/Automattic/wp-calypso/pull/11822/files#diff-246b58e6f07d7dce525ca439d82eccf9R40. Less chances of one of these calls getting deleted and it shows the intent (always prevent default behaviour of the event) better.